src/writer/serve.py

```
@@ -275,10 +276,21 @@ async def import_zip(file: UploadFile = File(...)):
                         raise HTTPException(status_code=413, detail=f"File too large. Max file size: {MAX_FILE_SIZE}")
                     tmp.write(chunk)
                 tmp_path = tmp.name
-            await app_runner.import_zip(tmp_path)
-            os.remove(tmp_path)
-        except ValueError:
-            raise HTTPException(status_code=400, detail="Invalid upload.")
+            
+            try:
+                await app_runner.import_zip(tmp_path)
+            except ValueError as e:
+                raise HTTPException(status_code=400, detail=str(e))
```

Ruff (0.14.4)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)